### PR TITLE
Fix PT-1345 vulnerabilities

### PR DESCRIPTION
**Pull request by yvon-dblg**
- Cleanup
  - Remove gitbook
  - Remove hakken
  - Remove other unused / deprecated files
- Add server stuffs to it's own directory
  - Support CloudFront build
  - Better CSP handling
- Update app & test
  - Remove global variable to use the config
  - Update some require to import/export
  - Fix route when using index.html as an entry
  - Move tidepool logo (selection done in webpack)
  - Remove dev/prod selection in require (done in webpack)
  - Update Intl doc
  - Include package tidepool-standard-action directly (no need a dep for one little function)
  - Add polyfills
- Update build & test system
  - webpack dev server: must be explicitly set
  - More simple docker builds
  - Rework artifact.sh part (not well name anymore...)
  - Bump SOUP versions
  - Update npm script targets
  - Units test with Firefox too.
- Update eslint config (most of the problems are set to warning for now)
- Update documentation
- Add package-lock.json

Require validations of:
- https://github.com/mdblp/ci-toolbox/pull/25

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -3,42 +3,115 @@
...
+    "jsx-a11y",
+    "moment-utc",
+    "lodash",
+    "import"
+  ],
+  "extends": [
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:jsx-a11y/recommended",
+    "plugin:react/recommended",
```
Comment (yvon-dblg) [inline .eslintrc:107]:
```suggestion
```

File: server/start.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+set -e
+npm run build-config
+exec node server.js
```
Comment (CedricPerotto) [inline server/start.sh:4]: This bash file is used for starting service on EC2 instances, so we need to source a config file, ensure node version through nvm (just like previous start.sh did).

Just keep the previous file
```suggestion
#!/bin/bash -eu

. "${NVM_DIR}/nvm.sh"
. version.sh

nvm ls "${START_NODE_VERSION}" > /dev/null || { echo "ERROR: Node version ${START_NODE_VERSION} not installed"; exit 1; }
nvm use --delete-prefix "${START_NODE_VERSION}"

. config/env.sh

npm run build-config
exec node server
```

File: server/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+COPY . .
+RUN \
+  chown -v node /app/dist/index.html \
+  && chown -v node /app/dist
+
+FROM buildServer AS production
+USER node
+ENV nexus_token="invalid"
+ENTRYPOINT [ "/bin/sh" ]
+CMD [ "start.sh" ]
```
Comment (CedricPerotto) [inline server/Dockerfile:16]: See my comment on the start.sh file. So we can't use it as CMD here : we don't have nvm (and don't care about it), we don't want to source missing env file. 

```suggestion
CMD [ "sh", "-c", "npm run build-config && npm run server" ]
```

File: Dockerfile [hunk 4]
```diff
@@ -1,127 +1,26 @@
-### Stage 0 - Base image
-FROM node:10.15.3-alpine as base
-WORKDIR /app
-RUN \
-  mkdir -p dist node_modules \
-  && chown -R node:node . \
-  && apk --no-cache  update \
-  && apk --no-cache  upgrade \
-  && npm config set unsafe-perm true \
+# Build for developers through Docker:
```
Comment (CedricPerotto) [inline Dockerfile:1]: I don't see the interest for this Dockerfile. 

And as it remains at the root of the project  it is also confusing : we could think that it is the current "production ready" Dockerfile for blip (whereas it is in server/Dockerfile).

File: index.ejs [hunk 5]
```diff
@@ -1,26 +1,21 @@
...
   </div>
   <!-- config -->
   <!-- Start of support Zendesk Widget script -->
-  <script id="ze-snippet" type="text/javascript" src="https://static.zdassets.com/ekr/snippet.js?key=d1aad8b3-67d8-4f44-9d87-9e8a0ed8149d"></script>
   <!-- End of support Zendesk Widget script -->
   <!-- Start of Tracker Code -->
-  <script type="text/javascript" src="matomo.js"></script>
   <!-- End of Tracker Code -->
+  <!-- Start of Crowdin -->
+  <!-- End of Crowdin -->
```
Comment (CedricPerotto) [inline index.ejs:19]: According to Crowdin doc:
It's important to add the Crowdin In-Context script before any other scripts, at the <head> section of your document. Make sure this script is included on every page with localizable text. Optionally, you can include this JavaScript snippet only when the pseudo-language is active.

**Pull request merged**